### PR TITLE
Cpfed 3850 close menu nav loses focus

### PR DESCRIPTION
**Pull request by kelsS**
---
name: Close any open menus when nav loses focus
labels: "feature, ready for review"
---

## pfe-navigation

Added feature to close any open menu dropdowns when the entire nav loses focus so that keyboard users cannot get to the rest of the page content with the nav still open.

### Testing instructions

1. Pull down changes
2. Tab through nav and onto the page content with any dropdown menu open
3. The dropdown menu should close when the last item in the nav loses focus and the focus is on the page content and not the nav.

### Merging

Please **squash** when merging and ensure your commit message uses [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) formatting.

### CODE REVIEW

File: elements/pfe-navigation/src/pfe-navigation.js [hunk 1]
```diff
@@ -195,7 +196,190 @@ class PfeNavigation extends PFElement {
         });
       });
     }
-  }
+
+    // Last link visually
```
Comment (wesruv) [inline elements/pfe-navigation/src/pfe-navigation.js:200]: This is a lot of commented out code?

File: elements/pfe-navigation/src/pfe-navigation.js [hunk 2]
```diff
@@ -838,6 +1023,18 @@ class PfeNavigation extends PFElement {
     // @todo/bug: figure out why this event listener only fires once you have tabbed into the menu but not if you have just clicked open menu items with a mouse click on Firefox - functions properly on Chrome
     this.addEventListener("keydown", this._generalKeyboardListener);
 
+    // @todo: check to see if this is the best place for these variables to be defined or should they go in the constructor?
```
Comment (wesruv) [inline elements/pfe-navigation/src/pfe-navigation.js:1026]: Looks like you're not using these vars anywhere but here, so here seems fine?

**Review by wesruv — APPROVED** [review]
Feel like the commented code could go, but otherwise it looks good.

I haven't tested it yet, but we can handle any issues as bugs, looks sound to me!

File: elements/pfe-navigation/src/pfe-navigation.js (hunk 1 above)
Comment (kelsS) [reply elements/pfe-navigation/src/pfe-navigation.js]: @wesruv thanks for catching that, that was from my original testing and working through the feature. Got that stale code removed.

**Pull request merged**